src/Core/Form/IdentifiableObject/DataHandler/AttributeGroupFormDataHandler.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Command\AddAttributeGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\Command\EditAttributeGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupId;
+use PrestaShop\PrestaShop\Core\Domain\AttributeGroup\ValueObject\AttributeGroupType;
+
+/**
+ * Handles data of submitted Attribute Group form.
+ */
+final class AttributeGroupFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(CommandBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data)
+    {
+        /** @var AttributeGroupId $attributeGroupId */
+        $attributeGroupId = $this->commandBus->handle(new AddAttributeGroupCommand(
+            $data['name'],
+            $data['public_name'],
+            (new AttributeGroupType($data['group_type'])),
+            $data['shop_association'] ?? []
```

Comment:
```suggestion
            $data['shop_association'] ?? []
```
Once the constructor requires this parameter I think you'll need to handle a fallback when multishop is disabled (because the field won't be present in the form). @zuk3975 encountered the same problematic here https://github.com/PrestaShop/PrestaShop/pull/31728/files#r1170874772

I have an idea for improvement, but until it's done you should use the same fallback `$this->shopContext->getContextShopIds()` (you may need Julius' PR to be merged to get this new method in the interface, we'll try and have it merged quickly not to block you too long)

Reply:
So I just keep $data['shop_association'] in both add and update without checking anything right?